apps/src/code-studio/components/LabContainer.jsx

```diff
@@ -16,19 +16,44 @@ import moduleStyles from './LabContainer.module.scss';
 import i18n from '@cdo/locale';
 import FontAwesome from '@cdo/apps/templates/FontAwesome';
 import ErrorBoundary from './ErrorBoundary';
+import {levelsForLessonId} from '@cdo/apps/code-studio/progressReduxSelectors';
+import MusicView from '@cdo/apps/music/views/MusicView';
+import StandaloneVideo2 from '@cdo/apps/standaloneVideo2/StandaloneVideo2';
+import ProjectContainer from '@cdo/apps/labs/projects/ProjectContainer';
+import {getStandaloneProjectId} from '@cdo/apps/labs/projects/utils';
 
-const LabContainer = ({children, onError}) => {
+const LabContainer = ({onError}) => {
   const isLabLoading = useSelector(state => state.lab.isLoading);
   const isPageError = useSelector(state => state.lab.isPageError);
 
   const overlayStyle = isLabLoading
     ? moduleStyles.showingBlock
     : moduleStyles.fadeInBlock;
 
+  const currentApp = useSelector(
+    state =>
+      levelsForLessonId(state.progress, state.progress.currentLessonId).find(
+        level => level.isCurrentLevel
+      ).app
+  );
+
   return (
     <ErrorBoundary fallback={<ErrorFallbackPage />} onError={onError}>
       <div id="lab-container" className={moduleStyles.labContainer}>
-        {children}
+        <ProjectContainer channelId={getStandaloneProjectId()}>
+          <div
+            style={{
+              width: '100%',
+              height: '100%',
+              visibility: currentApp !== 'music' && 'hidden',
```

Comment:
why are we using visibility here rather than conditionally rendering?

Reply:
I tried using conditional rendering but then ran into a problem I also encountered in earlier explorations: we don't seem to have a clean way to tear down Blockly and then recreate it.  If we stop rendering the lab, and then start again, we'll attempt to recreate Blockly, but hit a bunch of errors because it has left things behind on the window.